### ISSUE TITLE
Issue with development mode

### ISSUE THREAD

**Issue by savan77** [opencv/cvat#201]
Hi,

I am using CVAT in development mode. Until I created the first task everything was working fine. But then I tried to create another task but it takes forever to create a task. In fact, I have to click on 'Cancel' because there is no response. I thought the changes I made in code repo might have caused this, so I deleted everything and build the repo again by cloning from Github. Again, when I create a new task it takes forever. For instance, I clicked on 'Submit' 5 minutes before I started creating this issue but it still says "Successful request! Creating..".

**Comment by nmanovic** [comment]
@savan77 ,

Could you please send us logs? If you use CVAT without docker you need to archive "logs" directory.

**Comment by savan77** [comment]
@nmanovic 

Thanks for super fast reply, as usual. I have uploaded log file here: https://geekybeeai-my.sharepoint.com/:u:/g/personal/savan_visalpara_geekyb_com/EYQzn1aBbS1Jh1GDY7Bg4N4BsJQGv6FtlsNS8GQ1IQKWng?e=Y1vC99

There is only one file: cvat_server.log

**Comment by nmanovic** [comment]
@savan77 ,

Tasks are created by workers. You need to launch them as well. Please look at the file to see what should be started in parallel with the server: https://github.com/opencv/cvat/blob/develop/supervisord.conf

worker_default is used to create tasks, worker_low is used to run TF annotation process...

**Comment by savan77** [comment]
I am running CVAT- Debugging and I guess it starts all the required tasks. Also, it worked fine initially.

**Comment by savan77** [comment]
Here is the problem.

I am creating a task as usual. But when I click on 'Submit' the following pop up stays as it as forever.
![screenshot from 2018-11-19 16-35-51](https://user-images.githubusercontent.com/16082167/48703396-8a188380-ec19-11e8-8579-d3e91f899cc3.png)

Finally, when i click on 'cancel' and reload the page :
![screenshot from 2018-11-19 16-35-33](https://user-images.githubusercontent.com/16082167/48703418-a3b9cb00-ec19-11e8-8a81-afa93b8be1f2.png)

**Comment by nmanovic** [comment]
@savan77 ,

there is a known problem with VS code. It doesn't restart some processes if you stop and start debugging again.  Just run "CVAT RQ - default" one more time. 

Each time when you finish debugging close VS code, open it and run "CVAT- Debugging". In this scenario it works always and will start all necessary processes.

It is a pain for me as well. I don't know how to workaround the issue with VS Code.

**Comment by savan77** [comment]
@nmanovic 
when I run 'CVAT RQ- default' again it throws following error.
cvat.apps.engine.models.DoesNotExist: Task matching query does not exist.

**Comment by savan77** [comment]
Thanks for your time. I will take care of it. I was planning to send a PR for reading labels from text or csv file.

**Comment by VesninEN** [comment]
I'm just setup docker on Windows10. Restart. 
run in windows powershel these commands:
docker-compose build
docker-compose up -d 

Open localhost:8080, register and press button "Create new Task".
After filling fields in Task Configuration I cant't do anything - button "Submit" didn't take any effect (I only see big red licence text).
![2019-03-07_084030_cr](https://user-images.githubusercontent.com/29756588/53934950-a996c780-40b5-11e9-9161-06dc3fb887f9.png)

**Comment by satyavardhan626** [comment]
Hi @nmanovic 

Running Cvat in development mode without using Docker.
While creating Task it is showing 500 error.
Do we need to run anything Apart from server????


And i found this Solution
command=%(ENV_HOME)s/wait-for-it.sh %(ENV_CVAT_POSTGRES_HOST)s:5432 -t 0 -- bash -ic \
    "rm -f /tmp/cvat-server/httpd.pid && /usr/bin/python3 ~/manage.py migrate && \
    /usr/bin/python3 ~/manage.py collectstatic --no-input && \
    exec /usr/bin/python3 $HOME/manage.py runmodwsgi --log-to-terminal --port 8080 \
    --limit-request-body 1073741824 --log-level INFO --include-file ~/mod_wsgi.conf \
    %(ENV_DJANGO_MODWSGI_EXTRA_ARGS)s --locale %(ENV_LC_ALL)s \
    --server-root /tmp/cvat-server"

environment=SSH_AUTH_SOCK="/tmp/ssh-agent.sock"


But where and how to run this Command??????   
![Screenshot (48)](https://user-images.githubusercontent.com/40227677/93236594-a4cbcd00-f79c-11ea-928f-7be9c7bf6a94.png)

**Comment by satyavardhan626** [comment]
Hi @nmanovic 

Running Cvat in development mode without using Docker.
While creating Task it is showing 500 error.
Do we need to run anything Apart from server????


And i found this Solution
command=%(ENV_HOME)s/wait-for-it.sh %(ENV_CVAT_POSTGRES_HOST)s:5432 -t 0 -- bash -ic \
    "rm -f /tmp/cvat-server/httpd.pid && /usr/bin/python3 ~/manage.py migrate && \
    /usr/bin/python3 ~/manage.py collectstatic --no-input && \
    exec /usr/bin/python3 $HOME/manage.py runmodwsgi --log-to-terminal --port 8080 \
    --limit-request-body 1073741824 --log-level INFO --include-file ~/mod_wsgi.conf \
    %(ENV_DJANGO_MODWSGI_EXTRA_ARGS)s --locale %(ENV_LC_ALL)s \
    --server-root /tmp/cvat-server"

environment=SSH_AUTH_SOCK="/tmp/ssh-agent.sock"


But where and how to run this Command??????   
![Screenshot (48)](https://user-images.githubusercontent.com/40227677/93236594-a4cbcd00-f79c-11ea-928f-7be9c7bf6a94.png)


![Capture](https://user-images.githubusercontent.com/40227677/93237212-6aaefb00-f79d-11ea-8957-45083b531067.PNG)

**Comment by nmanovic** [comment]
@satyavardhan626 , please read contribution guide: https://github.com/openvinotoolkit/cvat/blob/develop/CONTRIBUTING.md

**Comment by satyavardhan626** [comment]
> @satyavardhan626 , please read contribution guide: https://github.com/openvinotoolkit/cvat/blob/develop/CONTRIBUTING.md

i have done all the steps ,But it is throwing 500 Error.

**Comment by nmanovic** [comment]
@satyavardhan626 , please ask on our gitter channel. Also you need to provide server exception stack for the issue.

**Comment by satyavardhan626** [comment]
> @satyavardhan626 , please ask on our gitter channel. Also you need to provide server exception stack for the issue.

Hi @nmanovic 
![django](https://user-images.githubusercontent.com/40227677/93303515-3416c600-f819-11ea-882e-c5cec139628e.PNG)
![rq-default](https://user-images.githubusercontent.com/40227677/93303518-3547f300-f819-11ea-87b3-186f9e48003c.PNG)

 ,
Running cvat in local without using Docker .

command :Select server: debug configuration and start it (F5) to run REST server and its workers

After hitting the above command cvat got opened and able to login aswell.
While creating Task it is showing 500 error.

raise ConnectionError(self._error_message(e))
redis.exceptions.ConnectionError: Error 111 connecting to localhost:6379. Connection refused.
[2020-09-16 05:33:24,014] ERROR django.server: "POST /api/v1/tasks/7/data HTTP/1.1" 500 181701
ERROR:django.server:"POST /api/v1/tasks/7/data HTTP/1.1" 500 181701
[2020-09-16 05:33:24,034] INFO django.server: "OPTIONS /api/v1/tasks/7 HTTP/1.1" 200 0
INFO:django.server:"OPTIONS /api/v1/tasks/7 HTTP/1.1" 200 0
[2020-09-16 05:33:24,783] INFO django.server: "DELETE /api/v1/tasks/7 HTTP/1.1" 204 0
INFO:django.server:"DELETE /api/v1/tasks/7 HTTP/1.1" 204 0

Do we need to run anything Apart from server????
How do we connect with Db in this proccess?

**Comment by nmanovic** [comment]
@satyavardhan626 , it is clear from the log that your redis server isn't running. Try read error messages and google them:

```
raise ConnectionError(self._error_message(e))
redis.exceptions.ConnectionError: Error 111 connecting to localhost:6379. Connection refused.
```

If you installed redis-server, it would start automatically. Did you follow CONTRIBUTION.md?

**Comment by satyavardhan626** [comment]
> @satyavardhan626 , it is clear from the log that your redis server isn't running. Try read error messages and google them:
> 
> ```
> raise ConnectionError(self._error_message(e))
> redis.exceptions.ConnectionError: Error 111 connecting to localhost:6379. Connection refused.
> ```
> 
> If you installed redis-server, it would start automatically. Did you follow CONTRIBUTION.md?

@nmanovic    redis-server installed already and followed all the process as per CONTRIBUTION.md.But still facing the issue.May be redis-server is not starting
